Ruby 2.15 install/build fails

We install ruby via the community plugin `zzbet.rbenv`. When it gets to the install task, it fails in Ubuntu 18.

Output:

```
Downloading ruby-2.1.5.tar.bz2...
-> https://cache.ruby-lang.org/pub/ruby/2.1/ruby-2.1.5.tar.bz2
Installing ruby-2.1.5...
WARNING: ruby-2.1.5 is past its end of life and is now unsupported.
It no longer receives bug fixes or critical security updates.
BUILD FAILED (Ubuntu 18.04 using ruby-build 20190320)
Inspect or clean up the working tree at /tmp/ruby-build.20190326160938.15756
Results logged to /tmp/ruby-build.20190326160938.15756.log
Last 10 log lines:
make[1]: *** [ext/openssl/all] Error 2
make[1]: *** Waiting for unfinished jobs....
compiling callback-7.c
compiling callback-8.c
installing default callback libraries
linking shared-object dl/callback.so
make[2]: Leaving directory '/tmp/ruby-build.20190326160938.15756/ruby-2.1.5/ext/dl/callback'
make[1]: Leaving directory '/tmp/ruby-build.20190326160938.15756/ruby-2.1.5'
uncommon.mk:180: recipe for target 'build-ext' failed
```

--- a/Vagrantfile
+++ b/Vagrantfile
@@ -2,7 +2,7 @@
 # vi: set ft=ruby :
 
 # Run the following command to set up this Vagrant box:
-# ansible-playbook site.yml -i inventory/dev --limit=vagrant
+# ansible-playbook site.yml --limit=vagrant
 
 # It takes around 20 minutes to complete the first setup/provision/deploy.
 # Your new local OFN instance will then be available at: http://localhost:8080
@@ -16,7 +16,7 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
   # please see the online documentation at vagrantup.com.
 
   config.vm.provider :virtualbox do |vbox|
-    config.vm.box = "ubuntu/xenial64"
+    config.vm.box = "ubuntu/bionic64"
 
     # Set box memory.
     vbox.customize ["modifyvm", :id, "--memory", "1792"]

--- a/bin/requirements.yml
+++ b/bin/requirements.yml
@@ -3,7 +3,7 @@
   version: 1.5.0
 
 - src: zzet.rbenv
-  version: 3.0.0
+  version: 3.4.2
 
 - src: jdauphant.nginx
   version: v2.8.1

--- a/playbooks/provision.yml
+++ b/playbooks/provision.yml
@@ -26,6 +26,9 @@
     - role: common # Install common apps and libraries, and setup shell.
       tags: common
 
+    - role: compatibility # Handle version-specific OS dependencies and configuration
+      tags: compatibility
+
     - role: language # Setup locale.
       tags: language
 

--- a/roles/common/tasks/main.yml
+++ b/roles/common/tasks/main.yml
@@ -17,7 +17,6 @@
       - python-pycurl
       - python-psycopg2
       - python3-psycopg2
-      - software-properties-common
 
       # unknown why or if needed
       # The playbooks run without these dependencies,
@@ -43,12 +42,6 @@
       - libxslt1-dev
       - memcached
   tags: packages
-
-- name: define additional build dependencies for Ruby 2.1.5 on Ubuntu 18
-  set_fact:
-    rbenv_extra_depends:
-      - libssl1.0-dev
-  when: ansible_distribution_major_version == '18'
 
 # Remove monit if still installed. The notified handlers will run only once
 # (when the main task triggers a change by removing the config file)

--- a/roles/common/tasks/main.yml
+++ b/roles/common/tasks/main.yml
@@ -17,7 +17,7 @@
       - python-pycurl
       - python-psycopg2
       - python3-psycopg2
-      - python-software-properties
+      - software-properties-common
 
       # unknown why or if needed
       # The playbooks run without these dependencies,

--- a/roles/common/tasks/main.yml
+++ b/roles/common/tasks/main.yml
@@ -44,6 +44,12 @@
       - memcached
   tags: packages
 
+- name: define additional build dependencies for Ruby 2.1.5 on Ubuntu 18
+  set_fact:
+    rbenv_extra_depends:
+      - libssl1.0-dev
+  when: ansible_distribution_major_version == '18'
+
 # Remove monit if still installed. The notified handlers will run only once
 # (when the main task triggers a change by removing the config file)
 - name: clean up monit

--- a/roles/compatibility/tasks/main.yml
+++ b/roles/compatibility/tasks/main.yml
@@ -1,0 +1,25 @@
+--- # Compatibility
+
+- name: install packages for Ubuntu 16
+  apt:
+    name: "{{ xenial_packages }}"
+  become: yes
+  vars:
+    xenial_packages:
+      - python-software-properties
+  when: ansible_distribution_major_version == '16'
+
+- name: install packages for Ubuntu 18
+  apt:
+    name: "{{ bionic_packages }}"
+  become: yes
+  vars:
+    bionic_packages:
+      - software-properties-common
+  when: ansible_distribution_major_version == '18'
+
+- name: define additional rbenv build dependencies for Ruby 2.1.5 on Ubuntu 18
+  set_fact:
+    rbenv_extra_depends:
+      - libssl1.0-dev
+  when: ansible_distribution_major_version == '18'

